File: app/src/main/java/us/shandian/giga/get/FinishedMission.java
package us.shandian.giga.get;

import androidx.annotation.NonNull;

public class FinishedMission extends Mission {

    public int serviceId = -1;
    public long streamUid = -1;
    public String qualityLabel = null;

    public FinishedMission() {
    }

    public FinishedMission(@NonNull DownloadMission mission) {
        source = mission.source;
        length = mission.length;
        timestamp = mission.timestamp;
        kind = mission.kind;
        storage = mission.storage;
        serviceId = mission.serviceId;
        streamUid = mission.streamUid;
        qualityLabel = mission.qualityLabel;
    }

}

File: app/src/main/java/us/shandian/giga/get/FinishedMission.kt
package us.shandian.giga.get

class FinishedMission : Mission {
    constructor()
    constructor(mission: DownloadMission) {
        source = mission.source
        length = mission.length
        timestamp = mission.timestamp
        kind = mission.kind
        storage = mission.storage
    }
}
